### PR TITLE
Creates devices table and associations

### PR DESCRIPTION
**Pull request by mohinid**
## Notion card
https://www.notion.so/saeloun/Add-tables-for-personal-employment-Allocated-devices-Address-71ef49d0c38e4242bd2998120141372d

## Summary
1. Created devices table.
2. Added reference of User and Company with aliases issued_to and issued_by.
3. Added Validations
4. Added RSpecs
5. Updated RSpecs for User and Company.

## Preview
Employee Details Schema-updated

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

On Rails console as well as RSpecs Test run.

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

**Review by rohitjoshixyz — APPROVED** [review]
LGTM 🎉

File: app/models/device.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  # Device type values
+  enum device_type: { laptop: "laptop", mobile: "mobile" }
+
+  # Specifications values
+  store_accessor :specifications, :processor, :ram, :graphics
+
+  # Validations
+  after_initialize :set_default_specifications, if: :new_record?
+  validates :model, length: { maximum: 100 }
```
[the remark this answers is not in the document]
Comment (mohinid) [reply app/models/device.rb:40]: Hi, what's the reason?

[the remark this answers is not in the document]
Comment (mohinid) [reply app/models/device.rb:40]: @keshavbiswa @rohitjoshixyz Naming it as "Name"(Top level) as if I use model_name as column name, it gives this error.

Failure/Error: it { [is_expected.to](http://is_expected.to/) belong_to(:issued_by) }
     ActiveRecord::DangerousAttributeError:
       model_name is defined by Active Record. Check to make sure that you don't have an attribute or method with the same name.

File: app/models/device.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#  fk_rails_...  (company_id => companies.id)
+#  fk_rails_...  (user_id => users.id)
+#
+class Device < ApplicationRecord
+  # Associations
+  belongs_to :issued_by, class_name: "Company", foreign_key: "company_id"
+  belongs_to :issued_to, class_name: "User", foreign_key: "user_id"
+
+  # Device type values
+  enum device_type: { laptop: "laptop", mobile: "mobile" }
```
Comment (keshavbiswa) [inline app/models/device.rb:33]: Let's add specs for enums as well.

```ruby
describe "Enums" do
    it { is_expected.to define_enum_for(:device_type).with_values(laptop: "laptop", mobile: "mobile").backed_by_column_of_type(:string) }
  end
```

File: app/models/device.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohitjoshixyz) [reply app/models/device.rb]: Model attribute is confusing with rails models. We can rename it to `device_model`

[the remark this answers is not in the document]
Comment (mohinid) [reply app/models/device.rb]: Hey, as I mentioned. I have renamed it as "Name". Please check files once.

**Comment by rohitjoshixyz** [comment]
Oh okay so the name means the device's model. Sure I am fine with this 👍. PR is already approved from me✌🏻

**Comment by mohinid** [comment]
> Oh okay so the name means the device's model. Sure I am fine with this 👍. PR is already approved from me✌🏻

Yes. Thanks!

**Review by keshavbiswa — APPROVED** [review]
LGTM!

File: spec/models/device_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      end
+    end
+
+    describe "Enums" do
+      it {
+  expect(subject).to define_enum_for(:device_type).with_values(
+    laptop: "laptop",
+    mobile: "mobile").backed_by_column_of_type(:string)
+}
+    end
```
Comment (keshavbiswa) [inline spec/models/device_spec.rb:36]: Use `do` `end` instead of {} for multi line blocks.

**Pull request merged**